File: backend/src/main/java/com/soulh/model/DoctorVerification.java
package com.soulh.model;


import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;


@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@org.springframework.data.mongodb.core.mapping.Document
public class DoctorVerification {

    @org.springframework.data.annotation.Id
    
    private String id;

    
    
    private User doctor;

    
    private String registrationNumber;

    private String councilName;      // e.g. Medical Council of India
    private String certificateUrl;   // uploaded credential doc path
    private String governmentIdUrl;  // uploaded government ID path
    private String adminNotes;
    private LocalDate expiresAt;     // annual re-verification date

    
    
    @Builder.Default
    private VerificationStatus status = VerificationStatus.PENDING;

    
    private LocalDateTime submittedAt;

    private LocalDateTime updatedAt;

    
    protected void onSubmit() {
        submittedAt = LocalDateTime.now();
    }

    
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: node-backend/models/DoctorVerification.js
const mongoose = require('mongoose');

const DoctorVerificationSchema = new mongoose.Schema({
  doctor: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true,
    index: true
  },
  registrationNumber: {
    type: String,
    required: true
  },
  councilName: {
    type: String
  },
  certificateUrl: {
    type: String
  },
  governmentIdUrl: {
    type: String
  },
  adminNotes: {
    type: String
  },
  expiresAt: {
    type: Date
  },
  status: {
    type: String,
    enum: ['PENDING', 'APPROVED', 'REJECTED'],
    default: 'PENDING'
  },
  submittedAt: {
    type: Date,
    default: Date.now
  },
  updatedAt: {
    type: Date,
    default: Date.now
  }
});

// toJSON transform helper
DoctorVerificationSchema.set('toJSON', {
  transform: (doc, ret) => {
    ret.id = ret._id.toString();
    delete ret._id;
    delete ret.__v;
    return ret;
  }
});

module.exports = mongoose.model('DoctorVerification', DoctorVerificationSchema);
